tests/chainerx_tests/unit_tests/routines_tests/test_math.py

```diff
@@ -14,23 +12,23 @@
 
 class UnaryMathTestBase(object):
 
-    def setup(self, shape, dtype):
-        if numpy.dtype(dtype).kind != 'f':
+    def setup(self):
+        dtype = numpy.dtype(self.dtype)
+        if dtype.kind != 'f':
             self.skip_backward_test = True
             self.skip_double_backward_test = True
         if dtype == 'float16':
             self.check_forward_options.update({'rtol': 1e-3, 'atol': 1e-3})
             self.check_backward_options.update({'rtol': 1e-3, 'atol': 1e-3})
             self.check_double_backward_options.update(
                 {'rtol': 1e-2, 'atol': 1e-2})
-        self.shape = shape
-        self.dtype = dtype
 
     def generate_inputs(self):
-        shape = self.shape
-        dtype = self.dtype
-        a = numpy.random.uniform(-1, 1, shape).astype(dtype)
-        return a,
+        if self.input == 'random':
+            return numpy.random.uniform(-1, 1, self.shape).astype(self.dtype),
+        if self.input in ('inf', '-inf', 'nan'):
```

Comment:
How about parameterizing `float('inf')`, etc. instead of `'inf'`.?
Some tests may want to parameterize finite values such as `0`.
Also it's consistent with parameterization style of `scalar`.

Reply:
Fixed.